packages/blockchain-wallet-v4-frontend/src/modals/Trade/index.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+import { compose } from 'redux'
+import styled from 'styled-components'
+
+import { Button, Icon, Text } from 'blockchain-info-components'
+import Flyout, { duration, FlyoutChild } from 'components/Flyout'
+import { OptionRightActionRow } from 'components/Rows'
+import { ModalName } from 'data/types'
+import modalEnhancer from 'providers/ModalEnhancer'
+
+const Column = styled.div`
+  display: flex;
+  flex-direction: column;
+`
+const IconWrapper = styled.div`
+  position: absolute;
+  top: 40px;
+  right: 40px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: 50%;
+  border: none;
+  background-color: ${({ theme }) => theme.grey100};
+  height: 32px;
+  width: 32px;
+`
+const ContentContainer = styled(Column)`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  margin-top: 128px;
+
+  & > div {
+    width: 100%;
+  }
+`
+const ButtonWrapper = styled.div`
+  display: flex;
+  align-items: flex-end;
+  margin-top: 40px;
+  padding: 0 40px 0 32px;
+
+  & > :first-child {
+    margin-right: 16px;
+  }
+`
+
+const rows: Array<RowType> = [
+  {
+    description: () => (
+      <FormattedMessage
+        id='modals.trade.swap.description'
+        defaultMessage='Exchange Crypto for Another'
+      />
+    ),
+    handleClick: () => {},
+    header: () => <FormattedMessage id='buttons.swap' defaultMessage='Swap' />,
+    iconName: 'swap'
+  },
+  {
+    description: () => (
+      <FormattedMessage
+        id='modals.trade.send.description'
+        defaultMessage='Transfer to Another Wallet'
+      />
+    ),
+    handleClick: () => {},
+    header: () => <FormattedMessage id='buttons.send' defaultMessage='Send' />,
+    iconName: 'send'
+  },
+  {
+    description: () => (
+      <FormattedMessage
+        id='modals.trade.receive.description'
+        defaultMessage='Receive Crypto to Your Wallet'
+      />
+    ),
+    handleClick: () => {},
+    header: () => <FormattedMessage id='buttons.receive' defaultMessage='Receive' />,
+    iconName: 'receive'
+  },
+  {
+    description: () => (
+      <FormattedMessage
+        id='modals.trade.receive.description'
+        defaultMessage='Add Cash to Your Cash Account'
+      />
+    ),
+    handleClick: () => {},
+    header: () => <FormattedMessage id='buttons.deposit' defaultMessage='Deposit' />,
+    iconName: 'deposit'
+  },
+  {
+    description: () => (
+      <FormattedMessage
+        id='modals.trade.receive.description'
+        defaultMessage='Cash Out to Your Bank Account'
+      />
+    ),
+    handleClick: () => {},
+    header: () => <FormattedMessage id='buttons.withdraw' defaultMessage='Withdraw' />,
+    iconName: 'withdraw'
+  }
+]
```

Comment:
descriptions and headers are just formatted messages you do not need to make it as function could be just 

```
description:
      <FormattedMessage
        id='modals.trade.receive.description'
        defaultMessage='Cash Out to Your Bank Account'
      />,
...
header: <FormattedMessage id='buttons.withdraw' defaultMessage='Withdraw' />,
```